build/release.mk

```diff
@@ -66,8 +66,22 @@ endif
 	@# Download prepackaged plugins
 	mkdir -p tmpprepackaged
 	@cd tmpprepackaged && for plugin_package in $(PLUGIN_PACKAGES) ; do \
-		curl -O -L https://plugins-store.test.mattermost.com/release/$$plugin_package.tar.gz; \
-		curl -O -L https://plugins-store.test.mattermost.com/release/$$plugin_package.tar.gz.sig; \
+		curl -f -O -L https://plugins-store.test.mattermost.com/release/$$plugin_package.tar.gz; \
+		curl -f -O -L https://plugins-store.test.mattermost.com/release/$$plugin_package.tar.gz.sig; \
+	done
+
+	@# Import Mattermost plugin public key
+	gpg --import build/plugin_publickey.asc
+
+	@# Prepare prepackage plugins
+	@for plugin_package in $(PLUGIN_PACKAGES) ; do \
+		cp tmpprepackaged/$$plugin_package.tar.gz $(DIST_PATH)/prepackaged_plugins/$$plugin_package.tar.gz; \
+		cp tmpprepackaged/$$plugin_package.tar.gz.sig $(DIST_PATH)/prepackaged_plugins/$$plugin_package.tar.gz.sig; \
+		VERIFY_SIGNATURE_RESULT=`gpg --status-fd 1 --verify $(DIST_PATH)/prepackaged_plugins/$$plugin_package.tar.gz.sig $(DIST_PATH)/prepackaged_plugins/$$plugin_package.tar.gz 2>/dev/null | grep GOODSIG`; \
```

Comment:
Grepping for `GOOD SIG` seems brittle. In my environment, `gpg --verify` returns with an exit status when the signature fails. Can we rely on that instead, with the added benefit of getting the output of `gpg` when a failure occurs for any reason?

Reply:
Good call!